src/main/drivers/serial_softserial.c

```diff
@@ -254,8 +254,8 @@ serialPort_t *openSoftSerial(softSerialPortIndex_e portIndex, serialReceiveCallb
     if (options & SERIAL_BIDIR) {
         // If RX and TX pins are both assigned, we CAN use either with a timer.
         // However, for consistency with hardware UARTs, we only use TX pin,
-        // and this pin must have a timer.
-        if (!timerTx)
+        // and this pin must have a timer, and it should not be N-Channel.
+        if (!(timerTx && !(timerTx->output & TIMER_OUTPUT_N_CHANNEL)))
```

Comment:
I find `if (!timerTx || (timerTx->output & TIMER_OUTPUT_N_CHANNEL)) {`  easier to understand (but both are equivalent)

Reply:
Doesn't `if (!timerTx || (timerTx->output & TIMER_OUTPUT_N_CHANNEL))` causes hard fault when `timerTx` is `NULL`?

Reply:
No, it short circuits if the first term `!timerTx` is true.

Reply:
My bad 😓